terminal.go

```diff
@@ -935,14 +943,15 @@ func (g *LightningTerminal) shutdown() error {
 
 	g.wg.Wait()
 
-	// The lnd error channel is only used if we are actually running lnd in
-	// the same process.
-	if g.cfg.LndMode == ModeIntegrated {
-		err := <-g.lndErrChan
+	// Do we have any last errors to display?
+	select {
+	case err := <-g.errQueue.ChanOut():
```

Comment:
ok cool, so the assumption here is that the first error added to the queue is the one we want to display (right?)

Reply:
also looks like there are a few other places that could read from this channel first in which case the log here would not necessarily be the initial error.

Reply:
(mostly just commenting for my self to help with understanding)

Reply:
Hmm, the logic here is a bit tricky to understand, yes. I guess some further refactoring of the way we start/stop the subsystems would be a good thing.
But during normal operation we only shut down on 4 triggers:
 - The loop server errors out
 - There is an error in the new error queue
 - Lnd shut down (if with an error, the above will trigger first, if normal shutdown this will trigger)
 - The shutdown interceptor triggers

So in case 1, 2 and 4 there is no error in the queue before we shut down. So it's only possible that something happened during shutdown. But you're right, there might be more than one error, so I added a loop that logs all remaining errors and returns just the last one to the outer component (which will be logged as the causing error).